src/python/pants/init/engine_initializer.py

```diff
@@ -143,7 +149,9 @@ def run_goal_rules(
             if not goal_product.subsystem_cls.activated(union_membership):
                 raise GoalNotActivatedException(goal)
             # NB: Keep this in sync with the property `goal_param_types`.
-            params = Params(specs, self.console, workspace, env_name)
+            params = Params(
+                specs, self.console, workspace, env_name, CurrentExecutingGoal(goal, goal_product)
+            )
```

Comment:
regarding #10542 I think that we must still be able to provide goal specific `Params` even when running multiple goals in parallel and as such this "should" be OK, no?

Reply:
Yes, this is safe from a #10542 perspective. But it has the other problem I mentioned above: https://github.com/pantsbuild/pants/pull/17674#discussion_r1163271801 ... =(